crates/fuel-core/src/executor.rs

```diff
@@ -2198,48 +2369,36 @@ mod tests {
             }
 
             assert!(compare_coinbase_addresses(
-                Address::from([1u8; 32]),
-                Address::from([1u8; 32])
+                ContractId::from([1u8; 32]),
+                ContractId::from([1u8; 32])
             ));
             assert!(!compare_coinbase_addresses(
-                Address::from([9u8; 32]),
-                Address::from([1u8; 32])
+                ContractId::from([9u8; 32]),
+                ContractId::from([1u8; 32])
             ));
             assert!(!compare_coinbase_addresses(
-                Address::from([1u8; 32]),
-                Address::from([9u8; 32])
+                ContractId::from([1u8; 32]),
+                ContractId::from([9u8; 32])
             ));
             assert!(compare_coinbase_addresses(
-                Address::from([9u8; 32]),
-                Address::from([9u8; 32])
+                ContractId::from([9u8; 32]),
+                ContractId::from([9u8; 32])
             ));
         }
 
         #[test]
-        fn invalidate_is_not_first() {
-            let mint = Transaction::mint(TxPointer::new(Default::default(), 1), vec![]);
-
-            let mut block = Block::default();
-            *block.transactions_mut() = vec![mint.into()];
-            block.header_mut().recalculate_metadata();
-
-            let validator = Executor::test(Default::default(), Default::default());
-            let validation_err = validator
-                .execute_and_commit(ExecutionBlock::Validation(block), Default::default())
-                .expect_err("Expected error because coinbase if invalid");
-            assert!(matches!(
-                validation_err,
-                ExecutorError::CoinbaseIsNotFirstTransaction
-            ));
-        }
-
-        #[test]
-        fn invalidate_block_height() {
-            let mint =
-                Transaction::mint(TxPointer::new(1.into(), Default::default()), vec![]);
+        fn invalidate_is_not_last() {
+            let mint = Transaction::mint(
+                TxPointer::new(Default::default(), 1),
```

Comment:
Hmm I don't think the way we check that the mint is the last tx is very robust. If I just change the tx_index to 0 here, I don't get the `MintIsNotLastTransaction` error.

Reply:
Yeah, because it should be `MintHasUnexpectedIndex` error=) I added proper support for `MintIsNotLastTransaction`